homeassistant/components/modbus/sensor.py

```diff
@@ -142,11 +142,12 @@ async def async_setup_platform(
             _LOGGER.error("Error in sensor %s structure: %s", entry[CONF_NAME], err)
             continue
 
-        if entry[CONF_COUNT] * 2 != size:
+        regsize = entry.get(CONF_COUNT, 1) * 2
+        if regsize != size:
             _LOGGER.error(
-                "Structure size (%d bytes) mismatch registers count (%d words)",
+                "Structure size (%d bytes) mismatch registers size (%d bytes)",
```

Comment:
I recommend:
"Structure request %d bytes,  but %d registers have a size of %d bytes",

Changing from words to bytes is a good idea.